sig/datadog/profiling/stack_recorder.rbs

```diff
@@ -0,0 +1,27 @@
+module Datadog
+  module Profiling
+    class StackRecorder
+      @no_concurrent_synchronize_mutex: ::Thread::Mutex
+
+      def initialize: (cpu_time_enabled: bool, alloc_samples_enabled: bool) -> void
+
+      def self._native_initialize: (
+        Datadog::Profiling::StackRecorder recorder_instance,
+        bool cpu_time_enabled,
+        bool alloc_samples_enabled,
+      ) -> true
+
+      def serialize: () -> untyped
```

Comment:
Looking at `serialize!`, what can this return except for a `String`? Is that tied to the unhappy path?

Reply:
The signature of this method should be either a `[::Time, ::Time, ::String]` or a `nil`. But when I used `[::Time, ::Time, ::String]?`, steep started complaining that `exporter.rb` that uses this output (but checks if it is `nil` first) may have stuff as `nil`.

I'm not entirely sure what's wrong (and I'm guessing I could introduce a `Struct` to contain stuff instead of an array), but since I was timeboxing this effort, I left this for a later attempt :)